tests/filter_task_by_name.yml

```diff
@@ -0,0 +1,44 @@
+name: Filter Task by Name Job
+description: Job to test the filter task by name feature
+service: TES
+versions:
+  - 1.0.0
+  - 1.1.0
+tags:
+  - logical
+  - filter
+  - filter task by name test
+  - all
+jobs:
+  - name: create_task
+    description: Create a new TES task
+    endpoint: /tasks
+    operation: POST
+    request_body: |
+      {
+        "name": "e70dda48-ComplianceTest",     
+        "description": "Compliance Test",
+        "executors": [
+          {
+            "image": "alpine",
+            "command": [
+              "echo",
+              "hello"
+            ]
+          }
+        ]
+      }
+    response:
+      200:
+  - name: list_tasks
+    description: Retrieve the list of tasks tracked by the TES server
+    endpoint: /tasks
+    operation: GET
+    query_parameters:
+      - view: BASIC
+      - name_prefix: e70dda48-ComplianceTest
```

Comment:
Note that this is a prefix, so an even better test would be to submit two tasks, (1) one with a name matching the prefix, (2) one _starting_ with that same string but ending differently, and (3) one starting with some other string but ending with the prefix. And then your test would only pass if only the first two tasks are returned. But then you would need an array size check again. And/or a test that checks not for identity but a pattern in your `filter.value`.

Reply:
This is an excellent test. The only problem I can think of is implementing it. The current library `DotMap` needs an explicit path to extract the value. It cant traverse an array, we need to specify the array index to fetch. But since the size is small, the step can be repeated for now.
I suppose the design would be like this.
Create Task 1 (Store id1) -> Create Task 2 (Store id2)-> Create Task 3 (Store id3)-> Get Tasks (Filter1-id1 & id2 present, Filter2-Size==2)

Reply:
For the last test, could you please explain that? Are these patterns the wildcards?

Reply:
I meant that in the second task (only _starting_ with the prefix) we don't expect an exact match. So we couldn't use the current `filter.value` implementation for that check. Rather, we'd need something like this: `my_prefix*`, or - even more powerful - a regex: `^myprefix`

Reply:
I have added the `regex` feature. Earlier it was just checking prefix and equal value through `.startsWith()`. Now we can define any regex pattern.

Reply:
For the first two tests, checking the size of array is not possible. If the same test is run multiple times, the size of those tasks will not remain two. 

I have added those two tests to check the prefix and same string value.
I used different prefixes here (not as you told to use same prefix), because we cant retrieve latest two values in ListTasks as it is not sorted through timestamp (or maybe I am not aware). It could be possible that for multiple compliance runs, the top two values in ListTasks are actually the same task, but we will be checking two different conditions (prefix and same string) for it. So, I thought its better to just separate the tasks and let each task have its own check.

Reply:
That's fine. In principle though, you could check the difference of the size of an array at time point $t_0$ and $t_1$. However, given that there may be other jobs being submitted in the meantime, you could only check for $m = n(t_1) - n(t_0)$ being $>2$. But that's not important if you found another way of checking this.